ShoppingCart/HomePage.aspx.cs

```
@@ -0,0 +1,108 @@
+                myReader = myCommand.ExecuteReader();
+                int i = 0;
+                while(myReader.Read())
+                {
+                    i++;
+                }
+                myReader.Close();
+                SqlDataReader myReader1 = null;
+                myReader1 = myCommand.ExecuteReader();
+                productName = new Label[i];
+                productPrice = new Label[i];
+                productId = new Dictionary<string, string>();
+                for (int j=0;j<i;j++)
+                {
+                    myReader1.Read();
+                    productName[j] = new Label();
+                    productName[j].Text = myReader1["PName"].ToString();
+                    productPrice[j] = new Label();
+                    productPrice[j].Text = myReader1["Price"].ToString();
+                    productId[productName[j].Text] = myReader1["PId"].ToString();
+                    this.Form.Controls.Add(productName[j]);
+                    this.Form.Controls.Add(new LiteralControl("<br>"));
+                    this.Form.Controls.Add(productPrice[j]);
+                    this.Form.Controls.Add(new LiteralControl("<br>"));
+                    Button addToCart = new Button();
+                    addToCart.Text = "Add "+ productName[j].Text+ " to cart";
+                    addToCart.ID = productName[j].Text;
+                    addToCart.Font.Size = FontUnit.Point(10);
+                    addToCart.Click += new EventHandler(AddToCart);
+                    this.Form.Controls.Add(addToCart);
+                    this.Form.Controls.Add(new LiteralControl("<br>"));
+                    this.Form.Controls.Add(new LiteralControl("<br>"));
+                }
+                myReader1.Close();
+                myConnection.Close();
+                HttpContext.Current.Session["productId"] = productId;
+            }
+            catch(Exception exception)
+            {

```

Empty catch block??
what will happen when there is exception?